service/src/main/java/dk/kvalitetsit/hello/controller/HelloController.java

```diff
@@ -25,27 +29,56 @@ public HelloController(HelloService helloService) {
     }
 
     @Override
-    public ResponseEntity<HelloResponse> v1HelloGet(String name) {
+    public ResponseEntity<List<HelloResponse>> v1HelloGet(String name) {
         logger.debug("Enter GET hello.");
 
-        // Just for demonstrating error response. Actual validation should most likely be somewhere else.
-        if(name.equalsIgnoreCase("NOT_VALID")) {
-            throw new BadRequestException(DetailedError.DetailedErrorCodeEnum._10, "%s is not a valid name.".formatted(name));
+        List <HelloServiceOutput> serviceResponseList;
+
+        if (name == null) {
+            serviceResponseList = helloService.helloServiceGetAll();
+
+        }
+        else {
+            // Just for demonstrating error response. Actual validation should most likely be somewhere else.
+            if(name.equalsIgnoreCase("NOT_VALID")) {
+                throw new BadRequestException(DetailedError.DetailedErrorCodeEnum._10, "%s is not a valid name.".formatted(name));
+            }
+            var serviceInput = new HelloServiceInput(name);
+
+            serviceResponseList = helloService.helloServiceGetByName(serviceInput);
+        }
```

Comment:
Det er måske lige ved at være lidt på grænsen, men jeg synes egentlig de to forskellige måder controlleren reagerer på er forretningslogik. Det betyder at logikken om name er null eller ej bør flyttes til service klassen.